src/providers/README.md

```diff
@@ -0,0 +1,27 @@
+# @dojo/dgrid/providers
+
+A lightweight API is provided to use any underlying data format or store with dgrid.
+
+Because dgrid takes a reactive, unidirectional approach, data providers may update state or interpret state without interference. The entirety of the grid's state is driven by the object created by the data provider.
+
+## Features
+
+- An abstract [base class](../bases/DataProviderBase.ts) to extend from
+- Sort functionality
+
+## Implementation
+
+Extend `DataProviderBase` and implement `buildData`. Data providers should always leave intact the first generic, which allows consumers to indicate the structure of data that will be used.
+
+`buildData` has access to two instance properties:
+
+- `options`: Unique to each data provider, indicated through `DataProviderBase`'s second generic, passed to the constructor. Used to configure the data source
```

Comment:
Will probably annoy those that are particularly detailed that we have a sentence and then a phrase with no period at the end. So I'd suggest: Used to configure the data source and unique to each data provider, indicated through `DataProviderBase`'s second generic, passed to the constructor ?